framework/src/actions/CommonOutputAction.C

```diff
@@ -146,83 +147,101 @@ CommonOutputAction::CommonOutputAction(InputParameters params)
 #endif
 
-  // Only create a Console if screen output was not created
-  if (getParam<bool>("console") && !hasConsole())
-    create("Console");
-  else
-    _pars.set<bool>("console") = false;
+    // Only create a Console if screen output was not created
+    if (getParam<bool>("console") && !hasConsole())
+      create("Console");
+    else
+      _pars.set<bool>("console") = false;
 
-  if (getParam<bool>("csv"))
-    create("CSV");
+    if (getParam<bool>("csv"))
+      create("CSV");
 
-  if (getParam<bool>("xml"))
-    create("XMLOutput");
+    if (getParam<bool>("xml"))
+      create("XMLOutput");
 
-  if (getParam<bool>("json"))
-    create("JSON");
+    if (getParam<bool>("json"))
+      create("JSON");
 
 #ifdef LIBMESH_HAVE_VTK
-  if (getParam<bool>("vtk"))
-    create("VTK");
+    if (getParam<bool>("vtk"))
+      create("VTK");
 #else
-  if (getParam<bool>("vtk"))
-    mooseWarning("VTK output requested but not enabled through libMesh");
+    if (getParam<bool>("vtk"))
+      mooseWarning("VTK output requested but not enabled through libMesh");
 #endif
 
-  if (getParam<bool>("xda"))
-    create("XDA");
+    if (getParam<bool>("xda"))
+      create("XDA");
+
+    if (getParam<bool>("xdr"))
+      create("XDR");
+
+    if (getParam<bool>("checkpoint"))
+      create("Checkpoint");
 
-  if (getParam<bool>("xdr"))
-    create("XDR");
+    if (getParam<bool>("gmv"))
+      create("GMV");
 
-  if (getParam<bool>("checkpoint"))
-    create("Checkpoint");
+    if (getParam<bool>("tecplot"))
+      create("Tecplot");
 
-  if (getParam<bool>("gmv"))
-    create("GMV");
+    if (getParam<bool>("gnuplot"))
+      create("Gnuplot");
 
-  if (getParam<bool>("tecplot"))
-    create("Tecplot");
+    if (getParam<bool>("solution_history"))
+      create("SolutionHistory");
 
-  if (getParam<bool>("gnuplot"))
-    create("Gnuplot");
+    if (getParam<bool>("dofmap"))
+      create("DOFMap");
 
-  if (getParam<bool>("solution_history"))
-    create("SolutionHistory");
+    if (getParam<bool>("controls") || _app.getParam<bool>("show_controls"))
+      create("ControlOutput");
 
-  if (getParam<bool>("dofmap"))
-    create("DOFMap");
+    if (!_app.getParam<bool>("no_timing") &&
+        (getParam<bool>("perf_graph") || getParam<bool>("print_perf_log") ||
+         _app.getParam<bool>("timing")))
+      create("PerfGraphOutput");
 
-  if (getParam<bool>("controls") || _app.getParam<bool>("show_controls"))
-    create("ControlOutput");
+    if (!getParam<bool>("color"))
+      Moose::setColorConsole(false);
+  }
+  else if (_current_task == "add_output")
+  {
+    // More like remove output. This can't be done during the "common_output" task because the
+    // finite element problem is not yet constructed
 
-  if (!_app.getParam<bool>("no_timing") &&
-      (getParam<bool>("perf_graph") || getParam<bool>("print_perf_log") ||
-       _app.getParam<bool>("timing")))
-    create("PerfGraphOutput");
```

Comment:
What about this little guy? This will break `perf_graph=true`, correct?

Reply:
That's till present above :smile: The diff is just kind of weird

Reply:
Damn. I looked a few times. That was sneaky for me.